### PR TITLE
[`Llama + Mistral`] Add attention dropout

### LINKED ISSUE #26616: Any way we can get dropout added to modeling_llama.py?
### Feature request

Re: https://github.com/OpenAccess-AI-Collective/axolotl/issues/672

We currently lack any way to prevent overweighting of specific neurons, which becomes a problem when trying to finetune on limited datasets. Something like dropout should go a long way towards being able to develop better models at smaller parameter counts.

### Motivation

See above.

### Your contribution

I don't have familiarity with this codebase.

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
fixes #26616 by adding the `attention_dropout` attribute to the config and the logic in the code.

### CODE REVIEW

File: src/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -492,7 +493,7 @@ def forward(
         # TODO: llama does not have dropout in the config??
         # It is recommended to use dropout with FA according to the docs
         # when training.
-        dropout_rate = 0.0  # if not self.training else self.attn_dropout
+        dropout_rate = self.attention_dropout  # if not self.training else self.attn_dropout
```
Comment (amyeroberts) [inline src/transformers/models/llama/modeling_llama.py:496]: This should still be conditional on `self.training`

Comment (ArthurZucker) [reply src/transformers/models/llama/modeling_llama.py:496]: Yep I also did not implement the logic, will update soon 🤕

**Review by amyeroberts — APPROVED** [review]
Thanks for adding!
